### PR TITLE
[docs] Describe test.timeouts

### LINKED ISSUE #6010: [docs] Document the `test.timeouts` method
Now some of the timeouts can be set for an individual test:
```js
test
    .timeouts({
        pageLoadTimeout:    2000,
        pageRequestTimeout: 60000,
        ajaxRequestTimeout: 60000
    })
    ('...', async t => {
    ...
});
```
We should reflect this in docs. Also the `t.setPageLoadTimeout` should be marked as obsolete in favor of the way described above.

See related comment: https://github.com/DevExpress/testcafe/pull/5979#discussion_r582030501

### PR DESCRIPTION
**Pull request by DIRECTcut**
fixes #6010

### CODE REVIEW

File: docs/articles/documentation/reference/test-api/test/timeouts.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+fixture `My Fixture`
+    .timeouts({
+        pageLoadTimeout:    2000,
+        pageRequestTimeout: 60000,
+        ajaxRequestTimeout: 60000
+    })
+    .page `http://devexpress.github.io/testcafe/example`;
+
+test('My test' , async t => {
+})
```
Comment (wentwrong) [inline docs/articles/documentation/reference/test-api/test/timeouts.md:34]: Fixture doesn't have the `timeouts` method. Currently the timeouts can be specified only for an individual test (`test.timeouts`) and for the entire test run (using CLI/runner API/config file).

Comment (DIRECTcut) [reply docs/articles/documentation/reference/test-api/test/timeouts.md]: You're right, that's my bad. Fixed it

**Review by wentwrong — APPROVED** [review]

File: docs/articles/documentation/reference/test-api/test/timeouts.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+`pageLoadTimeout` *(optional)*    | number | The time (in milliseconds) passed after the `DOMContentLoaded` event, within which TestCafe waits for the `window.load` event to fire. See [pageLoadTimeout](../../configuration-file.md#pageloadtimeout).
+`pageRequestTimeout` *(optional)* | number | The time (in milliseconds) to wait for HTML pages. See [pageRequestTimeout](../../configuration-file.md#ajaxrequesttimeout).
+`ajaxRequestTimeout` *(optional)* | number | Wait time (in milliseconds) for fetch/XHR requests. See [ajaxRequestTimeout](../../configuration-file.md#ajaxrequesttimeout).
+
+```js
+fixture`My Fixture`
+    .page`http://devexpress.github.io/testcafe/example`;
+
+test
+    .timeouts({
```
Comment (miherlosev) [inline docs/articles/documentation/reference/test-api/test/timeouts.md:29]: Formatting: key values should be aligned:

```js
     .timeouts({
         pageLoadTimeout:    2000, // here
         pageRequestTimeout: 60000,
         ajaxRequestTimeout: 60000
    })
```

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: docs/articles/documentation/reference/test-api/test/timeouts.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        pageLoadTimeout: 2000,
```
Comment (DIRECTcut) [inline docs/articles/documentation/reference/test-api/test/timeouts.md:30]:
```suggestion
        pageLoadTimeout:     2000,
```

File: docs/articles/documentation/reference/test-api/test/timeouts.md (hunk 2 above)
Comment (DIRECTcut) [reply docs/articles/documentation/reference/test-api/test/timeouts.md:29]: Done

**Review by miherlosev — APPROVED** [review]

File: docs/articles/documentation/reference/test-api/test/timeouts.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+test.timeouts({
+    [pageLoadTimeout]
+    [, pageRequestTimeout]
+    [, ajaxRequestTimeout]
+}) → this
+```
+
+Parameter                         | Type   | Description
+--------------------------------- | ------ | ---------------------------------------------------------------------------
+`pageLoadTimeout` *(optional)*    | number | The time (in milliseconds) passed after the `DOMContentLoaded` event, within which TestCafe waits for the `window.load` event to fire. See [pageLoadTimeout](../../configuration-file.md#pageloadtimeout).
```
Comment (titerman) [inline docs/articles/documentation/reference/test-api/test/timeouts.md:20]: simpler language
```suggestion
`pageLoadTimeout` *(optional)*    | number | Maximum time (in milliseconds) between the`DOMContentLoaded` event and the `window.load` event. See [pageLoadTimeout](../../configuration-file.md#pageloadtimeout).
```

File: docs/articles/documentation/reference/test-api/test/timeouts.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+`pageRequestTimeout` *(optional)* | number | The time (in milliseconds) to wait for HTML pages. See [pageRequestTimeout](../../configuration-file.md#ajaxrequesttimeout).
```
Comment (titerman) [inline docs/articles/documentation/reference/test-api/test/timeouts.md:21]: You can't *wait* for an HTML page. You can wait for a response to your request.
```suggestion
`pageRequestTimeout` *(optional)* | number | Maximum web server response time (in milliseconds). See [pageRequestTimeout](../../configuration-file.md#ajaxrequesttimeout).
```

File: docs/articles/documentation/reference/test-api/test/timeouts.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+`ajaxRequestTimeout` *(optional)* | number | Wait time (in milliseconds) for fetch/XHR requests. See [ajaxRequestTimeout](../../configuration-file.md#ajaxrequesttimeout).
```
Comment (titerman) [inline docs/articles/documentation/reference/test-api/test/timeouts.md:22]:
```suggestion
`ajaxRequestTimeout` *(optional)* | number | Maximum time (in milliseconds) between a fetch/XHR request and the response. See [ajaxRequestTimeout](../../configuration-file.md#ajaxrequesttimeout).
```

File: docs/articles/documentation/reference/test-api/testcontroller/setpageloadtimeout.md [hunk 7]
```diff
@@ -32,4 +34,4 @@ test(`Page load timeout`, async t => {
 });
 ```
 
-> Note that browsers raise the `DOMContentLoaded` event after they load and parse the HTML document, while `window.load` is raised after the browser loads all stylesheets, images and subframes. This is why `window.load` fires after the `DOMContentLoaded` event with a certain delay.
+> Browsers raise the `DOMContentLoaded` event after they load and parse the HTML document, while `window.load` is raised after the browser loads all stylesheets, images and subframes. This is why `window.load` fires after the `DOMContentLoaded` event with a certain delay.
```
Comment (titerman) [inline docs/articles/documentation/reference/test-api/testcontroller/setpageloadtimeout.md:37]: a) This side note might seem irrelevant at first glance, so let's get straight to the point. 
b) I'd avoid using "the HTML document" to describe the HTML content of a page - it's not always true these days

```suggestion
> The `DOMContentLoaded` event always precedes the `window.load` event. It fires as soon as the browser loads and parses the HTML content of the page. The `window.load` event, on the other hand, waits for the browser to load all the style sheets, images and sub-frames.
```

**Review by titerman — CHANGES_REQUESTED** [review]

File: docs/articles/documentation/reference/test-api/testcontroller/setpageloadtimeout.md (hunk 7 above)
Comment (DIRECTcut) [reply docs/articles/documentation/reference/test-api/testcontroller/setpageloadtimeout.md:37]: Can an **event** really wait for anything?

```suggestion
> The `DOMContentLoaded` event always precedes the `window.load` event. It fires when the browser has loaded and parsed the HTML content of the page. The `window.load` event, on the other hand, fires when the browser has loaded all the style sheets, images and sub-frames.
```

File: docs/articles/documentation/reference/test-api/test/timeouts.md (hunk 5 above)
Comment (DIRECTcut) [reply docs/articles/documentation/reference/test-api/test/timeouts.md:21]: > You can't wait for an HTML page. You can wait for a response to your request.

`Wait for an HTML (to be served)` == `wait for a response to a request for an HTML page`

> Maximum web server response time

Doesn't indicate that only requests for webpages (when the browser requests HTML from the server) are affected. A fetch request (for example) is still made to a web server, yet has nothing to do with this option. 

```suggestion
`pageRequestTimeout` *(optional)* | number | Maximum time (in milliseconds) for the web server to serve an HTML page. See [pageRequestTimeout](../../configuration-file.md#ajaxrequesttimeout).
```

Gets a bit repetitive. So I'd say `to wait for HTML pages` is good enough.

File: docs/articles/documentation/reference/test-api/test/timeouts.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
+---
+layout: docs
+title: Test.timeouts Method
+permalink: /documentation/reference/test-api/test/timeouts.html
+---
+# Test.timeouts Method
+
+Sets timeouts for the test.
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply docs/articles/documentation/reference/test-api/test/timeouts.md:8]: A bit redundant in my opinion (given that the table below gives a detailed description for each property), but ok.

File: docs/articles/documentation/reference/test-api/test/timeouts.md (hunk 5 above)
Comment (titerman) [reply docs/articles/documentation/reference/test-api/test/timeouts.md:21]: Your new suggestion is better and less colloquial, in a good way tbh. Thanks!

**Review by titerman — APPROVED** [review]

File: docs/articles/documentation/reference/test-api/test/timeouts.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
+---
+layout: docs
+title: Test.timeouts Method
+permalink: /documentation/reference/test-api/test/timeouts.html
+---
+# Test.timeouts Method
+
+Customize timeout values for the requests performed during a particular test.
```
Comment (arminal) [inline docs/articles/documentation/reference/test-api/test/timeouts.md:8]:
```suggestion
Customize timeout values for requests performed during a test.
```

File: docs/articles/documentation/reference/test-api/testcontroller/setpageloadtimeout.md [hunk 10]
```diff
@@ -32,4 +34,4 @@ test(`Page load timeout`, async t => {
 });
 ```
 
-> Note that browsers raise the `DOMContentLoaded` event after they load and parse the HTML document, while `window.load` is raised after the browser loads all stylesheets, images and subframes. This is why `window.load` fires after the `DOMContentLoaded` event with a certain delay.
+> The `DOMContentLoaded` event always precedes the `window.load` event. It fires when the browser has loaded and parsed the HTML content of the page. The `window.load` event, on the other hand, fires when the browser has loaded all the style sheets, images and sub-frames.
```
Comment (arminal) [inline docs/articles/documentation/reference/test-api/testcontroller/setpageloadtimeout.md:37]:
```suggestion
> The `DOMContentLoaded` event always precedes the `window.load` event. It fires when the browser has loaded and parsed the HTML content of the page. The `window.load` event, on the other hand, fires when the browser has loaded all the style sheets, images, and sub-frames.
```

File: docs/articles/documentation/reference/test-api/test/timeouts.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    [pageLoadTimeout]
+    [, pageRequestTimeout]
+    [, ajaxRequestTimeout]
+}) → this
+```
+
+Parameter                         | Type   | Description
+--------------------------------- | ------ | ---------------------------------------------------------------------------
+`pageLoadTimeout` *(optional)*    | number | Maximum time (in milliseconds) between the`DOMContentLoaded` event and the `window.load` event. See [pageLoadTimeout](../../configuration-file.md#pageloadtimeout).
+`pageRequestTimeout` *(optional)* | number | Maximum time (in milliseconds) for the web server to serve an HTML page. See [pageRequestTimeout](../../configuration-file.md#ajaxrequesttimeout).
```
Comment (DIRECTcut) [inline docs/articles/documentation/reference/test-api/test/timeouts.md:21]:
```suggestion
`pageRequestTimeout` *(optional)* | number | Maximum time (in milliseconds) for the web server to serve an HTML page. See [pageRequestTimeout](../../configuration-file.md#pagerequesttimeout).
```
